### PR TITLE
Sort imports

### LINKED ISSUE #12088: Sort imports?
I don't think we're currently sorting our imports. Black doesn't support sorting, but Ruff does when the `I` rule is enabled. Should we enable it (as part of our pre-commit hooks)?

### PR DESCRIPTION
**Pull request by cbrnr**
Fixes #12088.

### CODE REVIEW

**Comment by hoechenberger** [comment]
image

😳

I guess .... LGTM if green!! 😅😅😅😅

**Review by hoechenberger — APPROVED** [review]

**Comment by cbrnr** [comment]
@hoechenberger did you not manually check every change?

**Comment by larsoner** [comment]
> Is this real and caused by this PR?

See test code that will need to be adjusted:

https://github.com/mne-tools/mne-python/blob/fdaeb86206d334242bcbcfd09460c970b33e4dcd/mne/tests/test_import_nesting.py#L139-L151

**Comment by hoechenberger** [comment]
> @hoechenberger did you not manually check every change?

Almost done 🤠🤡

**Comment by cbrnr** [comment]
Docs fail to build: https://app.circleci.com/pipelines/github/mne-tools/mne-python/21327/workflows/4e4ee041-9cab-4be1-bec2-bceb73e0d8d5/jobs/60319?invite=true#step-140-5569_85 (`ImportError: Failed to import any of the following Qt binding modules: PyQt5, PySide2`).

**Comment by larsoner** [comment]
Oddly enoughI think it's actually due to a change in import order, specifically in [`mne/viz/backends/_qt.py`](https://github.com/mne-tools/mne-python/pull/12097/files#diff-58efdba68d9d5da5cbc9f12e29e7973ec9462cd17c0a8c0860d5ab055f751679) having the `matplotlib` import *before* the PyVista (which uses `qtpy`) breaks stuff. I'll push a commit that I think should fix it

**Comment by hoechenberger** [comment]
> Oddly enoughI think it's actually due to a change in import order

Ha! This is crazy 🤠

File: mne/viz/backends/_qt.py [hunk 1]
```diff
@@ -12,8 +12,10 @@
 import weakref
 from contextlib import contextmanager
 
+from qtpy import API_NAME  # noqa: F401, isort: skip
```
[the remark this answers is not in the document]
Comment (hoechenberger) [reply mne/viz/backends/_qt.py:15]: I'm trolling here, but … imagine this was a lazy import … we'd still be waiting for the side-effect 😅😅😅

File: examples/simulation/plot_stc_metrics.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 """
 ===============================================
 Compare simulated and estimated source activity
-===============================================
+===============================================.
```
Comment (larsoner) [inline examples/simulation/plot_stc_metrics.py:4]: Changes like this (messing up the title underlining) are what's breaking CircleCI

**Pull request merged**